telegram/src/main/kotlin/com/github/kotlintelegrambot/network/multipart/MultipartBodyFactory.kt

```diff
@@ -44,22 +41,21 @@ internal class MultipartBodyFactory(private val gson: Gson) {
         return listOfNotNull(chatIdPart, mediaGroupPart, disableNotificationPart, replyToMessageIdPart, allowSendingWithoutReplyPart) + filesParts
     }
 
-    private fun MediaGroup.takeFiles(): List<Pair<File, String>> = medias.flatMap { groupableMedia ->
-        when {
-            groupableMedia is InputMediaPhoto && groupableMedia.media is TelegramFile.ByFile -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.IMAGE
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.media is TelegramFile.ByFile && groupableMedia.thumb != null -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.VIDEO,
-                groupableMedia.thumb.file to MediaTypeConstants.IMAGE
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.media is TelegramFile.ByFile -> listOf(
-                groupableMedia.media.file to MediaTypeConstants.VIDEO
-            )
-            groupableMedia is InputMediaVideo && groupableMedia.thumb != null -> listOf(
-                groupableMedia.thumb.file to MediaTypeConstants.IMAGE
-            )
-            else -> emptyList()
+    private fun MediaGroup.takeFilesParts(): List<MultipartBody.Part> = medias.flatMap { groupableMedia ->
+        val mediaType = when (groupableMedia) {
+            is InputMediaPhoto -> MediaTypeConstants.IMAGE
+            is InputMediaVideo -> MediaTypeConstants.VIDEO
+            is InputMediaAudio -> MediaTypeConstants.AUDIO
+            else -> null
         }
+        mutableListOf<MultipartBody.Part>().apply {
+            when (val media = groupableMedia.media) {
+                is TelegramFile.ByFile -> add(media.file.toMultipartBodyPart(mediaType = mediaType))
+                is TelegramFile.ByByteArray -> add(media.fileBytes.toMultipartBodyPart(partName = media.filename!!, mediaType = mediaType))
+            }
+            if (groupableMedia is InputMediaVideo && groupableMedia.thumb != null) {
+                add(groupableMedia.thumb.file.toMultipartBodyPart(mediaType = MediaTypeConstants.IMAGE))
+            }
+        }.toList()
```

Comment:
Not important, but just out of curiosity sharing the `buildList` function (https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/build-list.html) that can be very handy for cases like this.

Reply:
Indeed, `buildList` perfectly suits here, but first we need to bump Kotlin version at least to 1.6. Otherwise, we'll have to mess with `@ExperimentalStdlibApi` annotations